### ISSUE TITLE
have_avdevice not defined

### ISSUE THREAD

**Issue by yakir12** [JuliaIO/VideoIO.jl#227]
Me too (saw this one #226). 

But this happens when I try `using` a package that is using `VidioIO`. Here's that package's repo: https://github.com/yakir12/Format2DB.jl
You'll see in its [Manifest](https://github.com/yakir12/Format2DB.jl/blob/939c4e1da33a1157894cd79419d5c6eb0b976c7c/Manifest.toml#L434) that `VideoIO` is at 0.6.11, so I'm not sure what the remedy would be. 

I even tried adding FFMPEG.jl to see if that would help, it didn't. FFMPEG is stuck at 0.2.4 (see [here](https://github.com/yakir12/Format2DB.jl/blob/939c4e1da33a1157894cd79419d5c6eb0b976c7c/Manifest.toml#L140)), so maybe that's the problem? But `Pkg` says it's `VideoIO` that set it to that version:
```julia
(Format2DB) pkg> add FFMPEG@0.3.0
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package FFMPEG [c87230d0]:
 FFMPEG [c87230d0] log:
 ├─possible versions are: [0.1.0, 0.2.0-0.2.4, 0.3.0] or uninstalled
 ├─restricted to versions 0.3.0 by an explicit requirement, leaving only versions 0.3.0
 └─restricted by compatibility requirements with VideoIO [d6d074c3] to versions: 0.2.0-0.2.4 — no versions left
   └─VideoIO [d6d074c3] log:
     ├─possible versions are: [0.4.0, 0.5.0-0.5.6, 0.6.0-0.6.11] or uninstalled
     └─restricted to versions 0.6.11 by an explicit requirement, leaving only versions 0.6.11
```

**Comment by yakir12** [comment]
OK, I've narrowed it down, but not sure hoe to fix it:
In `Format2DB`'s manifest the `VideoIO` version is 0.6.11 --> good.
But if I `]add https://github.com/yakir12/Format2DB.jl` to some clean stack, and check in that generated `Manifest.toml` file, `VideoIO`'s version is 0.6.10 --> bad.

No idea why...

**Comment by yakir12** [comment]
Adding `VideoIO@0.6.11` helped my actual case as well, so I can continue working, but not sure what all of this meant/means. Hope this is helpful somehow...

**Comment by SimonDanisch** [comment]
https://github.com/JuliaIO/FFMPEG.jl/issues/24

**Comment by ianshmean** [comment]
We might want to correct the FFMPEG compat bound in the general registry retroactively for previous VideoIO versions to prevent this situation. It happened to me too. Even though I updated my main version to 0.6.11, other packages tried to use 0.6.10

**Comment by yakir12** [comment]
Wah.. I now realize that every time I update I break my package. I'll need to set the same limits as you've mentioned.

**Comment by ianshmean** [comment]
Yeah, the mistake  I made  was setting `FFMPEG = "≥ 0.2.0"` rather than `FFMPEG = "0.2"`.

I don't think there's  an easier fix than just limiting projects to >= VideoIO 0.6.11, as even if we updated the general registry, the Project.toml files associated with each erroneous VideoIO release 0.6.8-0.6.10 would allow `FFMPEG = "≥ 0.2.0"` and break.

If we have any more reports of this I'll think more, but for now I'd say just do this for any project with VideoIO:

`VideoIO = "0.6.11"`

**Comment by MackHarry** [comment]
I also get this error, but doing 

```Julia 

Julia 
] 
add VideoIO@0.6.11
``` 
did not help for me, I still get 

``` bash
┌ Info: Precompiling VideoIO [d6d074c3-1acf-5d4c-9a43-ef38773959a2]
└ @ Base loading.jl:1260
ERROR: LoadError: UndefVarError: have_avdevice not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./Base.jl:26
...
...
```
when trying `using VideoIO`. 

any hints welcome, since package looks really useful! thx in advance.

**Comment by SimonDanisch** [comment]
did you try running `]free VideoIO; up` ? I think all packages involved should be tagged by now..

**Comment by MackHarry** [comment]
strange, this results in: 
``` julia
pkg> free VideoIO
ERROR: expected package `VideoIO [d6d074c3]` to be pinned, tracking a path, or tracking a repository
```

**Comment by SimonDanisch** [comment]
maybe up is enough then...

**Comment by galenlynch** [comment]
> as even if we updated the general registry, the Project.toml files associated with each erroneous VideoIO release 0.6.8-0.6.10 would allow FFMPEG = "≥ 0.2.0" and break

Is that true? I thought the registry's `Compat.toml` overrides the package's `Project.toml` in terms of resolving which packages are compatible with eachother.

Just noticed that the `Compat.toml` for VideoIO.jl has two more errors that are lurking:
```
["0.6.1-0.6.7"]
BinaryProvider = "0.3.0-*"
FixedPointNumbers = "0.3.0-*"
```

If for whatever reason people need a VideoIO version in that range, then those unconstrained upper bounds will someday be the cause of an error.

**Comment by galenlynch** [comment]
And I think these too:

```
["0.5.5-0.6.0"]
FixedPointNumbers = "0.3-0"

["0.6.0"]
BinaryProvider = "0.3-0"
```

**Comment by caleb-allen** [comment]
while `up VideoIO` alone upgraded to `v0.6.10` and was still broken, using `add VideoIO@0.6.11` did the trick for me

**Comment by ianshmean** [comment]
Should be fixed with v0.7.0 being released
